src/receta.py

```diff
@@ -0,0 +1,65 @@
+class Receta:
+    '''Clase que alberga los datos necesarios para guardar una receta correctamente.'''
+
+    def __init__(self, nombre, ingredientes, cantidades, duracion, dificultad, elaboracion):
+        self.nombre = nombre
+        self.ingredientes = list(ingredientes)
+        self.cantidades = list(cantidades)
+        self.duracion = duracion
+        self.dificultad = dificultad
+        self.elaboracion = elaboracion
+        self.carta_receta = ''
+
+    def kcals_maximas(self):
+        ''' Método que calcula el aporte calórico total de la receta a partir de sus ingredientes.
+            En este momento no se tendrá en cuenta la elaboración de la receta, ya que esta
+            puede influir en el numero de calorias.'''
+        kcals_max = 0
+        contador = 0
+        for ing in self.ingredientes:
+            kcals_max += ing.kcals_totales(self.cantidades[contador])
+            contador += 1
+        return round(kcals_max,2)
+
+    def hidratos_maximos(self):
+        ''' Método que calcula el número de hidratos de la receta a partir de sus ingredientes.'''
+        hidratos_max = 0
+        contador = 0
+        for ing in self.ingredientes:
```

Comment:
Agh. ¿No conoces map-reduce?

Reply:
conozco el metodo reduce pero no se como meter aquí el map ya que como tenemos que acceder a la función de cálculo de calorias de cada ingrediente y a su vez pasarle como parámetro la cantidad de dicho ingrediente, no consigo no utilizar un bucle for para los ingredientes y un contador para ir recorriendo a su vez las cantidades.